include/complex_quda.h

```diff
@@ -511,6 +519,14 @@ struct complex <float> : public cuFloatComplex
       return *this;
     }
 
+  __host__ __device__
```

Comment:
I don't think these can be further templated.  Note that there is a generic `complex<ValueType>` base class that implements operations like `-=` and `*=` etc., but there are also the specializations `complex<float>` and `complex<double>` that build upon cuComplex.h.  These latter specializations need to be fully reimplemented, hence the functions you refer to are written using explicit `float` and `double` types.  Agree?